### DISCUSSION TITLE
Run tasks in different terminals

### DISCUSSION THREAD

**Discussion by ivan-kleshnin** [go-task/task#1268] [Q&A] [answered]
Hi. Sorry for potentially stupid question, I'm far from the devops field...

### Is it possible to setup Taskfile to run tasks in different terminals?

The main goal is to have log outputs from each task visually separated.
Another goal is to run different processes in parallel and be able to interact with them (at least ^-c them) independently. 
I'm talking about dev machine and dev. experience. And finally, ideally, it should be VSCode / Webstorm inline, builtin terminals (not OS terminals).

Sounds like a common convenience thing, but it's suprisingly hard to google a whole solution.
- https://askubuntu.com/questions/46627/how-can-i-make-a-script-that-opens-terminal-windows-and-executes-commands-in-the
- https://stackoverflow.com/questions/57600967/is-it-possible-to-open-a-new-terminal-inside-vscode-from-inside-a-script

I understand it's kinda a chicken | egg issue (run a taskfile in terminal which opens new terminals... 🤔) but I'm still curious if someone in the community had orchestrated something like that. Maybe I'm approaching this wrong – please advice an alternative 🙏

**Comment by andreynering** [comment, marked as answer]
Hi @ivan-kleshnin,

I honestly just open terminal windows/tabs manually and run each task on its own window/tab.

We don't have any plans to build a feature to support that. Since each system is different, it would be impossible to have a solution that works everywhere.

If the commands you linked works (I haven't tried them), nothing stops you from adding them to your Taskfile. Something like:

```yaml
version: '3'

tasks:
  echo: '{{if .EXTERNAL}}gnome-terminal -- {{end}} echo hello world'
```

```bash
task echo EXTERNAL=1
```

or:

```yaml
version: '3'

tasks:
  echo: '{{.PREFIX}} echo hello world'
```

```bash
task echo PREFIX="gnome-terminal -- "
```
